### PR TITLE
Documentation: Update debian setup, remove apt-key

### PR DESCRIPTION
**Pull request by joker234**
`apt-key` is deprecated and will be removed, see https://manpages.debian.org/bullseye/apt/apt-key.8.en.html#DESCRIPTION

Therefore I would propose to change the setup instructions on the docs. I left the “Deprecated Method” and “Migration Method” as it was, because I wasn't sure if this is supposed to stay for reference.

Further information:
 - https://wiki.debian.org/DebianRepository/UseThirdParty
 - https://www.linuxuprising.com/2021/01/apt-key-is-deprecated-how-to-add.html
   → “To import OpenPGP keys directly from a keyserver to a file in /usr/share/keyrings:”
 - https://askubuntu.com/a/1303285

### Changelog category (leave one):
- Documentation (changelog entry is not required)

### Documentation entry for user-facing changes

- [x] Documentation is written (mandatory for new features)

### CODE REVIEW

**Comment by joker234** [comment]
P.S.: I personally would suggest to add this file as well, but this isn't part of the PR, as preferences weren't set before.
Filename: `/etc/apt/preferences.d/clickhouse`
```
Package: clickhouse-*
Pin: origin "packages.clickhouse.com"
Pin-Priority: 800

Package: *
Pin: origin "packages.clickhouse.com"
Pin-Priority: -10
```

**Comment by alexey-milovidov** [comment]
@Felixoid, maybe you can help review?

**Comment by Felixoid** [comment]
Have you tested it on ubuntu 20.04?

```
docker run -it ubuntu:20.04 bash
root@3af95ce1f37b:/# apt-get update && apt-get install -y apt-transport-https ca-certificates dirmngr sudo && gpg --no-default-keyring --keyring /usr/share/keyrings/clickhouse-keyring.gpg --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys 8919F6BD2B48D754 ; echo "deb [signed-by=/usr/share/keyrings/clickhouse-keyring.gpg] https://packages.clickhouse.com/deb stable main" | sudo tee /etc/apt/sources.list.d/clickhouse.list && apt-get update
....
Updating certificates in /etc/ssl/certs...
0 added, 0 removed; done.
Running hooks in /etc/ca-certificates/update.d...
done.
gpg: keybox '/usr/share/keyrings/clickhouse-keyring.gpg' created
gpg: failed to create temporary file '/root/.gnupg/.#lk0x000055af57e84f40.3af95ce1f37b.2829': No such file or directory
gpg: connecting dirmngr at '/root/.gnupg/S.dirmngr' failed: No such file or directory
gpg: keyserver receive failed: No dirmngr
deb [signed-by=/usr/share/keyrings/clickhouse-keyring.gpg] https://packages.clickhouse.com/deb stable main
Hit:1 http://archive.ubuntu.com/ubuntu focal InRelease
Hit:2 http://security.ubuntu.com/ubuntu focal-security InRelease
Hit:3 http://archive.ubuntu.com/ubuntu focal-updates InRelease
Hit:4 http://archive.ubuntu.com/ubuntu focal-backports InRelease
Get:5 https://packages.clickhouse.com/deb stable InRelease [2484 B]
Err:5 https://packages.clickhouse.com/deb stable InRelease
  The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 3E4AD4719DDE9A38
Reading package lists... Done
W: GPG error: https://packages.clickhouse.com/deb stable InRelease: The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 3E4AD4719DDE9A38
E: The repository 'https://packages.clickhouse.com/deb stable InRelease' is not signed.
N: Updating from such a repository can't be done securely, and is therefore disabled by default.
N: See apt-secure(8) manpage for repository creation and user configuration details.
root@3af95ce1f37b:/# ls /usr/share/keyrings/clickhouse-keyring.gpg
/usr/share/keyrings/clickhouse-keyring.gpg
```

The current way to install is softly deprecated and still works. But the new one is incompatible with the current LTS, so merging it requires additional `if` logic in the script, so I propose to leave it as it is until the `apt-key` become obsolete.

**Comment by Felixoid** [comment]
After some research, I've found working commands for ubuntu 20.04:

```
apt-get update && \
  apt-get install -y apt-transport-https ca-certificates dirmngr sudo && \
  GNUPGHOME=$(mktemp -d) && \
  GNUPGHOME=$GNUPGHOME gpg --no-default-keyring --keyring /usr/share/keyrings/clickhouse-keyring.gpg --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys 8919F6BD2B48D754 && \
  rm -r "$GNUPGHOME" && \
  echo "deb [signed-by=/usr/share/keyrings/clickhouse-keyring.gpg] https://packages.clickhouse.com/deb stable main" | sudo tee /etc/apt/sources.list.d/clickhouse.list && \
  apt-get update
```

It works as well for 18.04. But not for 16.04, it requires additional `chmod`

```
done.
gpg: keyring `/tmp/tmp.g3KFdGdeFf/secring.gpg' created
gpg: keyring `/usr/share/keyrings/clickhouse-keyring.gpg' created
gpg: requesting key 2B48D754 from hkp server keyserver.ubuntu.com
gpg: /tmp/tmp.g3KFdGdeFf/trustdb.gpg: trustdb created
gpg: key 2B48D754: public key "ClickHouse Inc. Repositories Key <packages@clickhouse.com>" imported
gpg: no ultimately trusted keys found
gpg: Total number processed: 1
gpg:               imported: 1  (RSA: 1)
deb [signed-by=/usr/share/keyrings/clickhouse-keyring.gpg] https://packages.clickhouse.com/deb stable main
Hit:1 http://security.ubuntu.com/ubuntu xenial-security InRelease
Hit:2 http://archive.ubuntu.com/ubuntu xenial InRelease
Hit:3 http://archive.ubuntu.com/ubuntu xenial-updates InRelease
Hit:4 http://archive.ubuntu.com/ubuntu xenial-backports InRelease
Get:5 https://packages.clickhouse.com/deb stable InRelease [2484 B]
Ign:5 https://packages.clickhouse.com/deb stable InRelease
Get:6 https://packages.clickhouse.com/deb stable/main amd64 Packages [58.0 kB]
Fetched 60.5 kB in 0s (83.5 kB/s) 
Reading package lists... Done
W: GPG error: https://packages.clickhouse.com/deb stable InRelease: The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 3E4AD4719DDE9A38
W: The repository 'https://packages.clickhouse.com/deb stable InRelease' is not signed.
N: Data from such a repository can't be authenticated and is therefore potentially dangerous to use.
N: See apt-secure(8) manpage for repository creation and user configuration details.
root@35daf76f27d4:/# chmod +r /usr/share/keyrings/clickhouse-keyring.gpg
root@35daf76f27d4:/# apt-get update
Hit:1 http://security.ubuntu.com/ubuntu xenial-security InRelease
Hit:2 http://archive.ubuntu.com/ubuntu xenial InRelease
Hit:3 http://archive.ubuntu.com/ubuntu xenial-updates InRelease
Hit:4 http://archive.ubuntu.com/ubuntu xenial-backports InRelease
Get:5 https://packages.clickhouse.com/deb stable InRelease [2484 B]
Fetched 2484 B in 0s (5401 B/s)   
Reading package lists... Done
```

I've tested it on debian 9 and it works. Most probably, it will be broken for debian 8 because of the apt version, but both are already EOL according to the [page](https://wiki.debian.org/LTS)

I think if you'd change commands to the given, it's ok to merge. Feel free to reopen :+1:

File: docs/en/getting-started/install.md [hunk 1]
```diff
@@ -118,9 +118,9 @@ It is recommended to use official pre-compiled `deb` packages for Debian or Ubun
 #### Setup the Debian repository
 ``` bash
 sudo apt-get install -y apt-transport-https ca-certificates dirmngr
-sudo apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv 8919F6BD2B48D754
+sudo gpg --no-default-keyring --keyring /usr/share/keyrings/clickhouse-keyring.gpg --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys 8919F6BD2B48D754
```
Comment (Felixoid) [inline docs/en/getting-started/install.md:121]: The following snippet works in all debian from 9 to 11 and ubuntu LTS from 16.04 to 22.04

```suggestion
GNUPGHOME=$(mktemp -d)
sudo GNUPGHOME="$GNUPGHOME" gpg --no-default-keyring --keyring /usr/share/keyrings/clickhouse-keyring.gpg --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys 8919F6BD2B48D754
sudo rm -r "$GNUPGHOME"
sudo chmod +r /usr/share/keyrings/clickhouse-keyring.gpg
```

**Review by Felixoid — CHANGES_REQUESTED** [review]

**Comment by rfraposa** [comment]
Is this ready to be merged? Or are we waiting to resolve the conversation above?

**Review by Felixoid — APPROVED** [review]

**Pull request merged**